transmart-core-db/src/main/groovy/org/transmartproject/db/clinical/CountsWithThresholdService.groovy

```diff
@@ -0,0 +1,245 @@
+/* (c) Copyright 2017, tranSMART Foundation, Inc. */
+
+package org.transmartproject.db.clinical
+
+import groovy.transform.CompileStatic
+import org.transmartproject.core.config.CountsThresholdResource
+import org.transmartproject.core.multidimquery.AggregateDataResource
+import org.transmartproject.core.multidimquery.CrossTableResource
+import org.transmartproject.core.multidimquery.counts.Counts
+import org.transmartproject.core.multidimquery.crosstable.CrossTable
+import org.transmartproject.core.multidimquery.hypercube.Dimension
+import org.transmartproject.core.multidimquery.query.*
+import org.transmartproject.core.users.PatientDataAccessLevel
+import org.transmartproject.core.users.SimpleUser
+import org.transmartproject.core.users.User
+
+import java.util.function.Function
+import java.util.stream.Collectors
+
+import static org.transmartproject.core.users.PatientDataAccessLevel.COUNTS
+import static org.transmartproject.core.users.PatientDataAccessLevel.COUNTS_WITH_THRESHOLD
+
+/**
+ * Service to obfuscate patient counts data.
+ * Use threshold ({@link this.patientCountThreshold}) field below which exact counts are hidden.
+ * {@link this.BELOW_THRESHOLD_VALUE} used for both patient and observation count instead of original values then.
+ */
+@CompileStatic
+class CountsWithThresholdService implements AggregateDataResource, CrossTableResource, CountsThresholdResource {
+
+    public static final long BELOW_THRESHOLD_VALUE = -2
+    public static final Counts BELOW_THRESHOLD_COUNTS = new Counts(BELOW_THRESHOLD_VALUE, BELOW_THRESHOLD_VALUE)
+
+    @Delegate
+    AggregateDataResource aggregateDataResource
+    CrossTableResource crossTableResource
+
+    long patientCountThreshold = 0
```

Comment:
It is not using the configuration. This should be changed.

Reply:
it does. See spring configurations in `TransmartCoreGrailsPlugin` file. `patientCountThreshold = config.patientCountThreshold ?: 0`

Reply:
Ah, you are right!

Reply:
Yes, but come on, we never define services like that, why do it differently here?

Reply:
Despite following grails naming convention `CountsWithThresholdService` does not depend on a framework nor any core-db classes and could be easily moved to core-api. 
What do you think about that?

Reply:
Not really if you want to group code by functionality.